### PR TITLE
Remove unused LZ4 dependency

### PR DESCRIPTION
**Pull request by mavam**
###  :notebook_with_decorative_cover: Description

This PR removes the dependency on LZ4, which we no longer need.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit.

### CODE REVIEW

**Review by dominiklohmann — CHANGES_REQUESTED** [review]
Looks good to me, but I'm not a fan of the `as_bytes` header. The name really doesn't describe well what it does.

Maybe remove them outright and add this to `span.hpp` or `byte.hpp`:

```
  template <class Buffer,
            class = std::enable_if_t<sizeof(*std::data(std::declval<Buffer>())) == 1>>
  static auto make(const Buffer& buffer)
    -> decltype(std::data(buffer), std::size(buffer), span<const byte>{}) {
  return span{reinterpret_cast<const byte*>(std::data(buffer)), std::size(buffer)};
}
```

File: libvast/vast/as_bytes.hpp [hunk 1]
```diff
@@ -11,28 +11,27 @@
...
-#include "vast/compression.hpp"
-#include "vast/die.hpp"
+#include "vast/byte.hpp"
+#include "vast/detail/type_traits.hpp"
+#include "vast/span.hpp"
 
-namespace vast {
-namespace lz4 {
+#include <type_traits>
+#include <vector>
```
Comment (dominiklohmann) [inline libvast/vast/as_bytes.hpp:21]: This include should no longer be necessary.

File: libvast/vast/chunk.hpp [hunk 2]
```diff
@@ -124,6 +90,20 @@ class chunk final : public caf::ref_counted {
   /// Avoid the common mistake of binding ownership to a string view.
   static auto make(std::string_view&&) = delete;
 
+  /// Construct a chunk from a byte buffer by copying it.
+  /// @param buffer The byte buffer.
+  /// @returns A chunk pointer or `nullptr` on failure.
+  template <class Buffer>
+  static auto copy(const Buffer& buffer)
```
Comment (mavam) [inline libvast/vast/chunk.hpp:97]: This works for now. Once our chunks support allocators, we could offer a second argument that contains an allocator.

**Review by dominiklohmann — APPROVED** [review]
I've done the required fixups necessary to finalize this PR. @mavam and I reviewed each other here, so I am approving his work, and he is approving my work by the means of merging. Let's wait for CI with merging, though.

**Pull request merged**